dataRetriever/factory/resolverscontainer/metaResolversContainerFactory.go

```diff
@@ -308,13 +309,25 @@ func (mrcf *metaResolversContainerFactory) generateTrieNodesResolvers() error {
 	resolversSlice := make([]dataRetriever.Resolver, 0)
 
 	identifierTrieNodes := factory.AccountTrieNodesTopic + core.CommunicationIdentifierBetweenShards(core.MetachainShardId, core.MetachainShardId)
-	resolver, err := mrcf.createTrieNodesResolver(
-		identifierTrieNodes,
-		triesFactory.UserAccountTrie,
-		0,
-		mrcf.numTotalPeers,
-		core.MetachainShardId,
-	)
+
+	var resolver dataRetriever.Resolver
+	var err error
+	if mrcf.isSyncedLiteObserver {
```

Comment:
ugly, but this is a factory, nevertheless. Will remain like this, no way around this ATM.